src/utility_belt/debug.clj

```diff
@@ -0,0 +1,27 @@
+(ns utility-belt.debug
+  (:require
+    [clojure.tools.logging :as log]
+    [utility-belt.time :as time]))
+
+
+(defmacro time+
+  "Similar to `time` macro, but prints before/after markers and associated tag"
+  [tag body]
+  `(let [state# (atom nil)]
+     (printf "[%s]:start %s\n"  ~tag (time/now))
+     (printf "[%s] %s" ~tag (with-out-str
+                              (time (reset! state# (do ~body)))))
+     (printf "[%s]:end %s\n"  ~tag (time/now))
+     (deref state#)))
```

Comment:
this is cool - but why do we need the state atom? wouldn't a let be enough?

Reply:
Because `with-out-string` captures the output of the expression it wraps, but discards what the body returns. So this happens:

```clojure

(defn something [] :return-value)

(time (something) ; => :return-value 
; timing information is printed in the terminal

(with-out-string (time (something)) ; => "Elapsed time ...." 
; the :return-value is lost at this point
```

There's a clever way of going around this using `binding` (see here: https://clojuredocs.org/clojure.core/with-out-str#example-590664dde4b01f4add58fe9f) but since this is only for debugging purposes (not even benchmarking!) an atom is Good Enough :tm:

Also, normally you'd use the `log-time` variant, as it's thread safe. `time+` is only useful if you have max 2 nested calls which do not use threads, otherwise everything will get smashed together in the output.

So yeah, it's **very basic** but I keep reinventing it every couple of weeks.